eth/api_backend.go

```diff
@@ -56,7 +56,9 @@ func (b *EthAPIBackend) CurrentBlock() *types.Block {
 
 func (b *EthAPIBackend) SetHead(number uint64) {
 	b.eth.protocolManager.downloader.Cancel()
-	b.eth.blockchain.SetHead(number)
+	if err := b.eth.blockchain.SetHead(number); err != nil {
+		panic(err)
```

Comment:
It'd be better to add an error return value to this method and pass it on to the caller.